media/js/base/core-datalayer.js

```diff
@@ -25,6 +25,55 @@ if (typeof Mozilla == 'undefined') {
         return pageId ? pageId : pathName.replace(/\/[^\/]*/, '');
     };
 
+    /** Returns page type.
+    * @param {String} path - URL path name fallback if page ID does not exist.
+    * @return {String} GTM page type.
+    */
+    Analytics.getPageType = function(path) {
+        var pathName = path ? path : document.location.pathname;
+
+        if (/firefox\/(desktop|android|ios)\/$/.test(pathName)) {
+            return 'product page';
+        } else if (/firefox\/(hello|sync|accounts|private-browsing)\/$/.test(pathName)) {
+            return 'feature page';
+        } else {
+            return null;
+        }
+    }
+
+    /** Returns whether page is long scrolling.
+    * @param {String} path - URL path name fallback if page ID does not exist.
+    * @return {String} boolean.
+    */
+    Analytics.isLongScrollingPage = function(path) {
+        var pathName = path ? path : document.location.pathname;
+
+        if (/firefox\/(desktop|android|ios|hello|sync|private-browsing)\/$/.test(pathName)) {
```

Comment:
This also seems like something that won't stand the test of time. I'd like to discuss something more dynamic / robust for reporting page length more accurately.